### ISSUE TITLE
Topsoil Moisture Evaluator in Surface Mesh Visualization Files (Feature Request)

### ISSUE THREAD

**Issue by saubhagya-gatech** [amanzi/ats#282]
Similar to the water table evaluator, it would be beneficial to have a topsoil moisture evaluator that computes the average soil moisture content for a specified number of upper layers and stores it on the surface mesh. With transpiration and water table depth already available in surface visualization files, this addition would reduce the need to frequently save subsurface files, as surface files would contain most of the key hydrological quantities.

The top 5 cm of soil is particularly valuable for model evaluations, especially for comparisons with field data or satellite products like SMAP and SMAP-HydroBlock.

**Comment by ecoon** [comment]
Please add an equation, and list out any parameters/flags/options you expect this would accept.

**Comment by saubhagya-gatech** [comment]
The soil moisture content of the topsoil, θ (in percentage), can be calculated as:

θ = (100 / D) ∫[0 to D] φ(z) * S(z) dz

where:
- φ(z) is the porosity as a function of depth z,
- S(z) is the saturation index as a function of depth z,
- D is the total depth over which the integration is performed.

Parameters:
- dependency keys for saturation and porosity field
- number of layers - this will be layers at the top of the domain for which average is taken.

**Comment by saubhagya-gatech** [comment]
yes, [-] is good, need not to be percentage. Sure, we can provide depth. So, perhaps consider layers whole bottlom elevation is higher than the D?

**Comment by ecoon** [comment]
We can interpolate if D is inconsistent.  I mean, if you ask for D = 5 cm, but your dzs [cm] are { 2, 2, 2, ...}, then, assuming piecewise constant saturations, it's WC_0 + WC_1 + WC_2 / 2.

**Comment by saubhagya-gatech** [comment]
right, I was more worried about thick layers below top soil if some gives too high D value. Interpolation should be fine. By the way, we can also just directly use water content divided by volume, either equation is fine.
